sdk/storage/azure-storage-blob-changefeed/src/test/java/com/azure/storage/blob/changefeed/ChunkTest.groovy

```diff
@@ -0,0 +1,209 @@
+import com.azure.storage.blob.BlobContainerAsyncClient
+import com.azure.storage.blob.changefeed.implementation.models.BlobChangefeedEventWrapper
+import com.azure.storage.blob.changefeed.implementation.models.ChangefeedCursor
+
+import com.azure.storage.blob.changefeed.models.BlobChangefeedEvent
+import com.azure.storage.blob.changefeed.models.BlobChangefeedEventData
+import com.azure.storage.internal.avro.implementation.AvroObject
+import com.azure.storage.internal.avro.implementation.AvroReader
+import com.azure.storage.internal.avro.implementation.AvroReaderFactory
+import reactor.core.publisher.Flux
+import reactor.test.StepVerifier
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import static org.mockito.ArgumentMatchers.*
+import static org.mockito.Mockito.*
+
+class ChunkTest extends Specification {
+
+    BlobContainerAsyncClient mockContainer
+    BlobAsyncClient mockBlob
+    AvroReaderFactory mockAvroReaderFactory
+    AvroReader mockAvroReader
+    BlobLazyDownloaderFactory mockBlobLazyDownloaderFactory
+    BlobLazyDownloader mockBlobLazyDownloader
+
+    String chunkPath = "chunkPath"
+    ChangefeedCursor chunkCursor
+
+    List<BlobChangefeedEvent> mockEvents
+    List<Map<String, Object>> mockRecords
+    List<AvroObject> mockAvroObjects
+
+    def setup() {
+        setupEvents()
+        chunkCursor = new ChangefeedCursor("endTime", "segmentTime", "shardPath", chunkPath, 0, 0)
+
+        mockContainer = mock(BlobContainerAsyncClient.class)
+        mockBlob = mock(BlobAsyncClient.class)
+        mockAvroReaderFactory = mock(AvroReaderFactory.class)
+        mockAvroReader = mock(AvroReader.class)
+        mockBlobLazyDownloaderFactory = mock(BlobLazyDownloaderFactory.class)
+        mockBlobLazyDownloader = mock(BlobLazyDownloader.class)
+
+        when(mockContainer.getBlobAsyncClient(anyString()))
+            .thenReturn(mockBlob)
+        when(mockBlobLazyDownloaderFactory.getBlobLazyDownloader(any(BlobAsyncClient.class), anyLong(), anyLong()))
+            .thenReturn(mockBlobLazyDownloader)
+        /* The data returned by the lazy downloader does not matter since we're mocking avroObjects.  */
+        when(mockBlobLazyDownloader.download())
+            .thenReturn(Flux.empty())
+        when(mockAvroReader.readAvroObjects())
+            .thenReturn(Flux.fromIterable(mockAvroObjects))
+    }
+
+    /* Tests no user cursor. */
+    def "getEvents min"() {
+        setup:
+        when(mockAvroReaderFactory.getAvroReader(any(Flux.class)))
+            .thenReturn(mockAvroReader)
+
+        when:
+        ChunkFactory factory = new ChunkFactory(mockAvroReaderFactory, mockBlobLazyDownloaderFactory, mockContainer)
+        Chunk chunk = factory.getChunk(chunkPath, chunkCursor, 0, 0)
+        def sv = StepVerifier.create(chunk.getEvents().index())
+
+        then:
+        sv.expectNextMatches({ tuple2 -> this.&verifyWrapper(tuple2.getT2(), tuple2.getT1(), 1234, 0) })
```

Comment:
Why are you using a method reference instead of just calling verifyWrapper?

Reply:
Can't we get the blockOffset and blockIndex based off the flux index (tuple2.getT1())? Otherwise it seems like hard coding the blockOffset and blockIndex in the test like this is only really testing that we hard coded things the same way here and when generating the mock data

Reply:
Re: first comment, it's just invoking the function as a closure. https://stackoverflow.com/questions/1395166/invoke-method-as-closure

Reply:
I'm not sure I follow how invoking the method as a closure is different from just invoking the method in this case? Why not just use this.verifyWrapper?

Reply:
just doing normal method call. I'm not sure why I used the closures. I think I googled a thing and it told me to do this.